src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -131,6 +134,19 @@ private bool HasCallerMemberNameAttribute
                                                   && !HasCallerFilePathAttribute
                                                   && HasCallerMemberNameAttribute;
 
+        internal override int CallerArgumentExpressionParameterIndex
+        {
+            get
+            {
+                if (!HasCallerLineNumberAttribute && !HasCallerFilePathAttribute && !HasCallerMemberNameAttribute && HasCallerArgumentExpressionAttribute)
```

Comment:
>!HasCallerLineNumberAttribute && !HasCallerFilePathAttribute && !HasCallerMemberNameAttribute [](start = 20, length = 93)

An attribute is either present or not, the fact doesn't depend on whether other attributes are present.

Reply:
This was following the other attributes implementations:

https://github.com/dotnet/roslyn/blob/79cae7252e3bf9a34e1913c7b7a86f9ba167412a/src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs#L129-L135

But good point. At a first glance, it indeed feels redundant since the binder tries to read the attributes in the correct order. Which basically mean if we are reading this property while binding, we are already sure that other attributes doesn't exist.

Will check if removing them breaks anything.

Reply:
@AlekseyTs Looks like it's okay to remove them. Should I remove them from my code and also the existing code?

Reply:
>  Looks like it's okay to remove them. Should I remove them from my code and also the existing code?

Either way is fine with me. I think "cleaning up" the existing code is the right thing to do.

---
In reply to: [600287343](https://github.com/dotnet/roslyn/pull/51952#discussion_r600287343) [](ancestors = 600287343)